Automattic/wp-calypso
# Settings: add preview site link to Launch site card

- closes https://github.com/Automattic/dotcom-forge/issues/1188

#### Proposed Changes

* Create a new component `SitePreviewLink` that consists of a toggle and a non-editable text input with the preview URL.
* Add the `SitePreviewLink` to Settings > Launch Site if the site is `Business` or `Ecommerce`.
* The `SitePreviewLink` is behind a feature flag `dev/share-site-for-preview`.


#### Testing Instructions


* Create a `Business` or `Ecommerce` site and don't launch it.
* Go to `Settings`.
* Find the card `Launch site`.
* Observe a new toggle to generate a "Preview Link".
* Click on the toggle.
* Observe a new link is generated.
* Copy the link and open it on a new tab in an incognito tab.
* Observe you can see the site.
* Go back to settings and Toggle off the "Preview Link".
* Refresh the incognito tab and observe the site is not visible anymore.

**Screencast**



https://user-images.githubusercontent.com/779993/204843302-20329154-acb9-47f6-8909-b8b2e75b8883.mp4





**Screenshot**

<img width="747" alt="Screenshot 2022-11-22 at 13 01 07" src="https://user-images.githubusercontent.com/779993/203320301-8938ed69-f39a-4c5d-8347-ab107a0ec4f2.png">

| **Before** | **After** |
|---|---|
| <img width="733" alt="Screenshot 2022-11-21 at 10 20 00" src="https://user-images.githubusercontent.com/779993/203089373-52732ffd-b913-41c4-96a6-2b922ab35616.png"> | <img width="734" alt="Screenshot 2022-11-21 at 09 57 50" src="https://user-images.githubusercontent.com/779993/203089440-b6c8b667-9604-48c7-8498-d85d8c631adf.png">  |


#### Pre-merge Checklist





- Blocked by:
   - https://github.com/Automattic/dotcom-forge/issues/1187
